feat: quickstart by multi-app run file

Eileen-Yu opened the pull request:
go with https://github.com/diagridio/cli/pull/718

new workflow:
1. user run `dg project quickstart` to download the corresponding sample code
2. user is responsible for language dependency installation (npm install, etc.)
3. user run `dg dev run -f dev-<type>.yaml --project myproject` to create project, resources and set up tunnel 

~~The only question is how to enable the workflow by the multi-app run file?? maybe add a flag --enable-workflow for `dev run` cmd~~ resolved in https://github.com/diagridio/cli/pull/718

jjcollinge commented:
any reason not to call the app run file `dapr.yaml` as it customary in dapr?

famarting commented:
now the docs need to be updated after renaming all these run files https://github.com/diagridio/docs/pull/605 , can you do that @Eileen-Yu

famarting commented:
need to revert this commit https://github.com/diagridio/catalyst-quickstarts/pull/70/commits/73731708deb713db6bae6a7598dad46d76ea5145 so the docs are fine

famarting commented:
> need to revert this commit [7373170](https://github.com/diagridio/catalyst-quickstarts/commit/73731708deb713db6bae6a7598dad46d76ea5145) so the docs are fine

done

Eileen-Yu commented:
updated the yaml file name to be exactly the project name, as required here https://github.com/diagridio/cli/pull/718#issuecomment-2807456618

also updated the docs in https://github.com/diagridio/docs/pull/605
